### PR TITLE
Pull Request for VQ-VAE model on HipMRI data

### PR DESCRIPTION
**Pull request by Izen78**
Hello Teaching Team,

I chose task 7 which is creating a VQ-VAE/VQ-VAE2 model for the HipMRI data for prostate cancer. This task has been successful in creating a model that goes above a 0.6 threshold for the SSIM where there is a 99% rate in going above this threshold. The file structure in this PR is below:

- assets/: which contains all the images that is added into the README.md
- README.md: The README.md file documents the model's architecture, data pipeline and the results of generating the images.
- dataset.py: This file contains the conversion of Nifti files into training, validation and test dataloaders used for train.py and predict.py
- modules.py: This file contains the model architecture and all the classes for the components in creating the VQ-VAE model
- predict.py: This file contains inference of the model and calculates the SSIM and generates reconstructed images using the saved model
- train.py: This file contains the training and validation loops of the VQ-VAE model

Thanks,

Arpon Sarker (s4745413)

### CODE REVIEW

**Comment by YashMitttal** [comment]
Some additional feedback:

- Including graphs for training and validation loss, as well as SSIM, is excellent. However, make sure the plots are clear (i.e., the axes should be labelled appropriately).
- You mention resizing the images to 256x128, but it would be helpful to explain why this size was chosen and whether any other preprocessing (e.g., normalization, augmentation) was applied.
- It’s great that you included details about training (e.g., epochs, batch size, optimizer). However, I would like to see more explanation of your choice of optimizer (Adam) and learning rate. Why were these values selected? How did you tune them?

**Comment by shakes76** [comment]
Quality of output, loss dip ocurred and some inconsistencies -3
